packages/nova-react/src/eventing/nova-eventing-provider.tsx

```diff
@@ -4,10 +4,16 @@ import { InputType } from "@nova/types";
 import invariant from "invariant";
 
 // Initializing default with null to make sure providers are correctly placed in the tree
-const NovaEventingContext = React.createContext<NovaReactEventing | null>(null);
+const NovaEventingContext = React.createContext<INovaEventingContext>({});
+
+interface INovaEventingContext {
+  eventing?: NovaReactEventing;
+  unmountEventing?: NovaReactEventing;
+}
 
 interface NovaEventingProviderProps {
   eventing: NovaEventing;
+  unmountEventing?: NovaEventing; // Optional version of eventing to use when unmounting, defaults to eventing if not supplied
```

Comment:
Can we change the comment to a doc comment over the prop?

Reply:
Updated